src/server/pkg/storage/fileset/merge.go

```diff
@@ -112,6 +106,44 @@ func tagMergeFunc(w *Writer) mergeFunc {
 	}
 }
 
+var (
+	// ShardThreshold the size threshold for shard creation.
+	// (bryce) this will be configurable at some point.
+	ShardThreshold int64 = 1024 * chunk.MB
+)
+
+func shardMergeFunc(f func(r *index.PathRange) error) mergeFunc {
```

Comment:
a comment on what this function does, both what it's used for and the semantics it implements would be good.

Reply:
Sure.